staging/src/k8s.io/apiserver/pkg/endpoints/handlers/patch.go

```diff
@@ -301,25 +310,50 @@ type jsonPatcher struct {
 	fieldManager *fieldmanager.FieldManager
 }
 
-func (p *jsonPatcher) applyPatchToCurrentObject(currentObject runtime.Object) (runtime.Object, error) {
+func (p *jsonPatcher) applyPatchToCurrentObject(requestContext context.Context, currentObject runtime.Object) (runtime.Object, error) {
 	// Encode will convert & return a versioned object in JSON.
 	currentObjJS, err := runtime.Encode(p.codec, currentObject)
 	if err != nil {
 		return nil, err
 	}
 
 	// Apply the patch.
-	patchedObjJS, err := p.applyJSPatch(currentObjJS)
+	patchedObjJS, appliedStrictErrs, err := p.applyJSPatch(currentObjJS)
 	if err != nil {
 		return nil, err
 	}
 
 	// Construct the resulting typed, unversioned object.
 	objToUpdate := p.restPatcher.New()
 	if err := runtime.DecodeInto(p.codec, patchedObjJS, objToUpdate); err != nil {
-		return nil, errors.NewInvalid(schema.GroupKind{}, "", field.ErrorList{
-			field.Invalid(field.NewPath("patch"), string(patchedObjJS), err.Error()),
-		})
+		strictError, isStrictError := runtime.AsStrictDecodingError(err)
+		switch {
+		case !isStrictError:
+			// disregard any appliedStrictErrs, because it's an incomplete
+			// list of strict errors given that we don't know what fields were
+			// unknown because DecodeInto failed. Non-strict errors trump in this case.
+			return nil, errors.NewInvalid(schema.GroupKind{}, "", field.ErrorList{
+				field.Invalid(field.NewPath("patch"), string(patchedObjJS), err.Error()),
+			})
+		case p.validationDirective == metav1.FieldValidationStrict:
```

Comment:
put this case last and make it `default:`. we're only using `StrictSerializer` if the directive is Warn or Strict, so if we get a strict error, and we're not warning, it should be fatal

Reply:
An update on this, I wasn't able to remove the `returnUnknownFieldPaths` field from the unstructuredNegotiatedSerializer, because there are tests that use the serializer directly (i.e. not through the CRD) that didn't seem to make sense to introduce the concept of strict errors to for the scope of that test.

For example `apiextensions-apiserver/pkg/apiserver.TestHandlerConversionWithWatchCache`

Reply:
hrmmm... I didn't realize that logic was active unconditionally

At the point where we construct schemaCoercingDecoder and unstructuredSchemaCoercer (inside unstructuredNegotiatedSerializer#DecoderToVersion), we have the selected request-specific decoder passed in as a parameter, right?

Can we tell at that point based on the input decoder if strict decoding was selected or not, rather than unconditionally enabling it in unstructuredNegotiatedSerializer?